src/main/java/core/basesyntax/service/impl/DataConverterImpl.java

```diff
@@ -0,0 +1,24 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.model.Operation;
+import core.basesyntax.service.DataConverter;
+import java.util.List;
+
+public class DataConverterImpl implements DataConverter {
+    @Override
+    public List<FruitTransaction> convert(List<String> lines) {
+        return lines.stream()
+                .map(this::convertToFruitTransaction)
+                .toList();
+    }
+
+    private FruitTransaction convertToFruitTransaction(String line) {
+        String[] fields = line.split(",");

```

Comment:
There is no error handling for the case where the input line does not contain enough fields, which could lead to an ArrayIndexOutOfBoundsException.